app/frontend/wallet/shelley/shelley-trezor-crypto-provider.ts

```diff
@@ -122,17 +129,53 @@ const CardanoTrezorCryptoProvider = ({network, config}) => {
     }
   }
 
+  function poolCertToTrezorFormat(cert) {
+    return {
+      poolId: cert.poolKeyHashHex,
+      vrfKeyHash: cert.vrfKeyHashHex,
+      pledge: cert.pledgeStr,
+      cost: cert.costStr,
+      margin: {
+        numerator: cert.margin.numeratorStr,
+        denominator: cert.margin.denominatorStr,
+      },
+      rewardAccount: bech32.encode('stake', Buffer.from(cert.rewardAccountHex, 'hex')),
+      owners: cert.poolOwners.map((owner) => ({
+        ...(owner.stakingKeyHashHex && {
+          stakingKeyHash: owner.stakingKeyHashHex,
+        }),
+        ...(owner.stakingPath && {
+          stakingKeyPath: owner.stakingPath,
+          stakingKeyHash: undefined,
+        }),
+      })),
+      relays: cert.relays.map((relay) => ({
+        type: relay.type,
+        ...(relay.type === 0 && {
+          ipv4Address: relay.params.ipv4,
+          ipv6Address: relay.params.ipv6,
+        }),
+        ...(relay.type < 2 && {port: relay.params.portNumber}),
```

Comment:
comparing enums with `<` seems really hacky and fragile (what if new types of relays emerge which have a totally different structure?). What I would expect instead is having the enums represented by named constants and test whether the given relay type belongs to a specific collection (be it an array or set) of relay types

Reply:
@xdzurman

Reply:
This was initially done since the three types of relays shared the attributes and this saved a lot of lines and copypasting at the time, however I agree with refactoring it.

Basically, there's the:
1. "normal/basic" relay, which has type "0", port, ipv4 address, ipv6 address
2. "single host" relay, which has type 1, port, and dns hostname
and
3. "multi host" relay, which has type 2 and dns hostname